hazelcast/src/main/java/com/hazelcast/nio/ascii/TextDecoder.java

```diff
@@ -113,20 +120,34 @@ public TextDecoder(TcpIpConnection connection, TextEncoder encoder) {
         this.textCommandService = ioService.getTextCommandService();
         this.encoder = encoder;
         this.connection = connection;
-        this.memcacheEnabled = ioService.isMemcacheEnabled();
-        this.restEnabled = ioService.isRestEnabled();
-        this.healthcheckEnabled = ioService.isHealthcheckEnabled();
+        HazelcastProperties props = ioService.properties();
+
+        this.memcacheEnabled = props.getBoolean(MEMCACHE_ENABLED);
+        this.restEnabled = props.getBoolean(REST_ENABLED);
+        this.healthcheckEnabled = props.getBoolean(HTTP_HEALTHCHECK_ENABLED);
         this.logger = ioService.getLoggingService().getLogger(getClass());
     }
 
+    @Override
+    public void requestClose() {
+        initSrcBuffer();
```

Comment:
The source buffer is initialized when we request close and not when the handler is added? Is this correct?

Reply:
The wrong method is called here. The buffer should be initialized in the 'handlerAdded' method.